### PR TITLE
add more info to sanity

### PR DESCRIPTION
**Pull request by MelvinTo**

### CODE REVIEW

File: scripts/sanity_check.sh [hunk 1]
```diff
@@ -409,35 +451,45 @@ check_network() {
...
     case $1 in
-        -f | --fast )           shift
-                                FAST=true
-                                ;;
-        -h | --help )           usage
-                                exit
-                                ;;
-        * )                     usage
-                                exit 1
+    -f | --fast)
```
Comment (MelvinTo) [inline scripts/sanity_check.sh:477]: only format change

File: scripts/sanity_check.sh [hunk 2]
```diff
@@ -409,35 +451,45 @@ check_network() {
     curl localhost:8837/v1/config/interfaces -o /tmp/scc_interfaces &>/dev/null
     INTFS=$(cat /tmp/scc_interfaces | jq 'keys' | jq -r .[])
 
-    echo "Interface,Name,UUID,Enabled,IPv4,IPv6,Gateway,Gateway6,DNS" > /tmp/scc_csv
+    echo "Interface,Name,UUID,Enabled,IPv4,IPv6,Gateway,Gateway6,DNS" >/tmp/scc_csv
     for INTF in $INTFS; do
-        cat /tmp/scc_interfaces | jq -r ".[\"$INTF\"] | [\"$INTF\", .config.meta.name // \"\", .config.meta.uuid[0:8], .config.enabled, .state.ip4 // \"\", (.state.ip6 // [] | join(\",\")), .state.gateway // \"\", .state.gateway6 // \"\", (.state.dns // [] | join(\";\"))] | @csv" >> /tmp/scc_csv
+        cat /tmp/scc_interfaces | jq -r ".[\"$INTF\"] | [\"$INTF\", .config.meta.name // \"\", .config.meta.uuid[0:8], .config.enabled, .state.ip4 // \"\", (.state.ip6 // [] | join(\",\")), .state.gateway // \"\", .state.gateway6 // \"\", (.state.dns // [] | join(\";\"))] | @csv" >>/tmp/scc_csv
```
Comment (MelvinTo) [inline scripts/sanity_check.sh:456]: only format change

File: scripts/sanity_check.sh [hunk 3]
```diff
@@ -3,35 +3,52 @@
 check_cloud() {
     echo -n "  checking cloud access ... "
     curl_result=$(curl -w '%{http_code}' -Lks --connect-timeout 5 https://firewalla.encipher.io)
-    test $curl_result == '200' && echo OK || { echo "fail($curl_result)"; return 1; }
+    test $curl_result == '200' && echo OK || {
```
Comment (MelvinTo) [inline scripts/sanity_check.sh:6]: only format change

File: scripts/sanity_check.sh [hunk 4]
```diff
@@ -52,12 +69,10 @@ check_git() {
...
     pushd $repo_dir >/dev/null
     git_output=$(git status -uno --porcelain 2>&1)
-    if [[ -n "$git_output" || $? -ne 0 ]]
-    then
+    if [[ -n "$git_output" || $? -ne 0 ]]; then
         echo fail
         _rc=1
-    elif [[ $remote_origin != $(git remote -v | awk '/origin/ {print $2}' | uniq) ]]
-    then
+    elif [[ $remote_origin != $(git remote -v | awk '/origin/ {print $2}' | uniq) ]]; then
```
Comment (MelvinTo) [inline scripts/sanity_check.sh:75]: only format change

File: scripts/sanity_check.sh [hunk 5]
```diff
@@ -70,26 +85,35 @@ check_git() {
 check_each_system_service() {
     local SERVICE_NAME=$1
     local EXPECTED_STATUS=$2
-    local ACTUAL_STATUS=$(systemctl show -p SubState $SERVICE_NAME | sed 's/SubState=//')
-    printf "%20s %10s %10s\n" $SERVICE_NAME $EXPECTED_STATUS $ACTUAL_STATUS
+    local RESTART_TIMES=$(systemctl show $1 -p NRestarts | awk -F= '{print $2}')
+    local ACTUAL_STATUS=$(systemctl status $1 | grep 'Active: ' | sed 's=Active: ==')
+    printf "%20s %10s %5s %s\n" $SERVICE_NAME $EXPECTED_STATUS "$RESTART_TIMES" "$ACTUAL_STATUS"
```
Comment (MelvinTo) [inline scripts/sanity_check.sh:90]: add number of restart retries and how long this service has been active

File: scripts/sanity_check.sh [hunk 6]
```diff
@@ -409,35 +451,45 @@ check_network() {
...
+        cat /tmp/scc_interfaces | jq -r ".[\"$INTF\"] | [\"$INTF\", .config.meta.name // \"\", .config.meta.uuid[0:8], .config.enabled, .state.ip4 // \"\", (.state.ip6 // [] | join(\",\")), .state.gateway // \"\", .state.gateway6 // \"\", (.state.dns // [] | join(\";\"))] | @csv" >>/tmp/scc_csv
     done
     cat /tmp/scc_csv | column -t -s, | sed 's=\"==g'
     echo ""
     echo ""
 }
 
+check_dhcp() {
+    echo "---------------------- DHCP ------------------"
+    find /log/blog/ -mmin -120 -name "dhcp*log.gz" | sort | xargs zcat
```
Comment (jimyang2008) [inline scripts/sanity_check.sh:465]: To get CSV format of the output, and ignore key names


```
# with key names
jq -r '.uids=(.uids|join("|"))|.msg_types=(.msg_types|join("|")) | to_entries|map(.key), map(.value)|@csv'

# without key names
jq -r '.uids=(.uids|join("|"))|.msg_types=(.msg_types|join("|")) | to_entries|map(.value)|@csv'
```

**Pull request merged**